time.go

```diff
@@ -1,16 +1,26 @@
 package main
 
-import "time"
+import (
+	"time"
+)
 
 var fakeTime time.Time
+var now = time.Now
 
+//Now returns the current time
 func Now() time.Time {
 	if !fakeTime.IsZero() {
 		return fakeTime
 	}
-	return time.Now()
+	return now()
 }
```

Comment:
Thanks! time.go coverage is up! Just felt a little disconcerting, I'll put my thoughts in order to make the gdp code better.

The value of `Now()` is to make `IsSafetyHour()` easier to test. `IsSafetyHour()` is built into the business logic in `cli.go`.

If you can make `IsSafetyHour()` testable, coverage of `Now()` is not so important. For example, `fakeTime` and the `now` you added are very similar variables, although they have different names. I was wondering if I could eliminate this duplication. Also, `Now()` actually has a conditional branch and does not necessarily return the current time. I figured out how to separate the code for testing.

You can make `IsSafetyHour()` testable if you have a variable of either `fakeTime` or `now`. The pattern that allows the time generation process to be changed from the outside by making time.Time a variable, as in the article you put in the PR Description, is one of several time-related test patterns. We will consider it as a base for this time.

First, change the `IsSafetyHour()` to refer to the `now` variable directly in `IsSafetyHour()`. Then there will be no more calls to `Now()`.

```go
// cli.go
func IsSafetyHour() bool {
	return now().Hour() >= SafetyHourStart && now().Hour() < SafetyHourEnd
}
```

Now the `now` variable is responsible for returning the current time, as `Now()` does, and all the remaining code in time.go is for testing. The time.go and time_test.go are removed. And move the `now` variable to cli.go.

```go
// cli.go
var now = time.Now
```

And the function for the test to rewrite the time appears only in the test code. So it can be implemented as a test helper.

```go
// cli_test.go
func fakeNow(t *testing.T, fake time.Time) {
	t.Helper()
	t.Cleanup(func() {
		now = time.Now
	})
	now = func() time.Time {
		return fake
	}
}
```

In cli_test.go, replace the `Set()` call with `fakeNow()`.

Now all tests will succeed as before!

-----

What do you think about my idea? The Issue I made was in reference to test coverage of time.go, but the PR you sent me gives me an opportunity for a larger refactoring.

Reply:
> If you can make IsSafetyHour() testable, coverage of Now() is not so important.

I think so too. And your idea is so good to realize what I really wanted to do.
I fixed. [2211e59](https://github.com/Connehito/gdp/pull/29/commits/2211e594d409f1cbec00ef2141545997dfdc10b5)